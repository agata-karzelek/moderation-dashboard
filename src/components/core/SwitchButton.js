import Switch from "@material-ui/core/Switch";
import SwitchButtonStyle from "../../style/switchButton";

const SwitchButton = SwitchButtonStyle(({ classes, ...props }) => {
  return (
    <Switch
      focusVisibleClassName={classes.focusVisible}
      disableRipple
      classes={{
        root: classes.root,
        switchBase: classes.switchBase,
        thumb: classes.thumb,
        track: classes.track,
        checked: classes.checked,
      }}
      {...props}
    />
  );
});

export default SwitchButton;
