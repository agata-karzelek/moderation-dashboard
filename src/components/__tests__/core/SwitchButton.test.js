import React from "react";
import { shallow } from "enzyme";
import SwitchButton from "../../core/SwitchButton";

describe("Test Cases for SwitchButton", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<SwitchButton />);
  });

  test("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
