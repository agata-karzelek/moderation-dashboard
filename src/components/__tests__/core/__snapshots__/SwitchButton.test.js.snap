// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Cases for SwitchButton Snapshot 1`] = `
<Component
  classes={
    Object {
      "checked": "Component-checked-5",
      "focusVisible": "Component-focusVisible-6",
      "root": "Component-root-1",
      "switchBase": "Component-switchBase-2",
      "thumb": "Component-thumb-3",
      "track": "Component-track-4",
    }
  }
/>
`;
